### PR TITLE
Further grb utils cleanup

### PR DESCRIPTION
**Pull request by pannarale**
This PR follows up on comments received in, e.g., https://github.com/gwastro/pycbc/pull/4877.

It simply removes the other calls to `resolve_url_to_file` in `grb_utils.py` and some remaining commented lines of code.  It also fixes a typo in a comment in `pycbc_multi_inspiral`.

This change affects: PyGRB

It was tested in producing [this webpage](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_sep2024_1/).

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: pycbc/workflow/grb_utils.py [hunk 1]
```diff
@@ -457,10 +457,9 @@ def build_veto_filelist(workflow):
     return veto_files
 
 
-def build_segment_filelist(workflow):
+def build_segment_filelist(seg_dir):
     """Construct a FileList instance containing all segments txt files"""
 
-    seg_dir = workflow.cp.get('workflow', 'segment-dir')
     file_names = ["bufferSeg.txt", "offSourceSeg.txt", "onSourceSeg.txt"]
```
Comment (spxiwh) [inline pycbc/workflow/grb_utils.py:463]: A suggestion here to combine these 3 files into one (these were horrible even in 2011).

Comment (spxiwh) [reply pycbc/workflow/grb_utils.py:463]: ... Probably not for this PR though.

**Review by spxiwh — APPROVED** [review]
LGTM. One suggestion for the future below, but happy for this to be merged as is.

File: pycbc/workflow/grb_utils.py (hunk 1 above)
Comment (pannarale) [reply pycbc/workflow/grb_utils.py:463]: I opened an issue https://github.com/gwastro/pycbc/issues/4887 for future PyGRB development.  Thanks!

**Pull request merged**